graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/QueryValidationResource.java

```diff
@@ -67,52 +68,54 @@ public QueryValidationResource(QueryValidationService queryValidationService) {
     @ApiOperation("Validate a search query")
     @NoAuditEvent("Only validating query structure, not changing any data")
     public ValidationResponseDTO validateQuery(
-            @ApiParam(name = "validationRequest") ValidationRequestDTO validationRequest,
-            @Context SearchUser searchUser
+            @ApiParam(name = "validationRequest") final ValidationRequestDTO validationRequest,
+            @Context final SearchUser searchUser
     ) {
 
         final ValidationRequest.Builder q = ValidationRequest.Builder.builder()
                 .query(validationRequest.query())
                 .timerange(validationRequest.timerange().orElse(defaultTimeRange()))
                 .streams(searchUser.streams().readableOrAllIfEmpty(validationRequest.streams()))
-                .parameters(resolveParameters(validationRequest));
+                .parameters(resolveParameters(validationRequest))
+                .validationMode(toMode(validationRequest.validationMode()));
 
         validationRequest.filter().ifPresent(q::filter);
 
         final ValidationResponse response = queryValidationService.validate(q.build());
         return ValidationResponseDTO.create(toStatus(response.status()), toExplanations(response));
     }
 
-    private ImmutableSet<Parameter> resolveParameters(ValidationRequestDTO validationRequest) {
+    private ValidationMode toMode(final ValidationModeDTO validationMode) {
+        switch (validationMode) {
+            case QUERY -> {return ValidationMode.QUERY;}
+            case SEARCH_FILTER -> {return ValidationMode.SEARCH_FILTER;}
+            default -> throw new IllegalArgumentException("Unexpected validation mode " + validationMode.name());
+        }
+    }
```

Comment:
1. I am not sure if having both ValidationMode and ValidationModeDTO, with the same enum values, is really needed and I vote for removal of one of them.
2. If you insist of keeping them, this code can be probably simplified with smth like ValidationMode.valueOf(validationMode.name()).

Reply:
This is done to strictly separate the DTOs used by our resources and internal implementation. But we can indeed simplify the conversion. Done that now. Thanks!